includes/class-salesforce.php

```diff
@@ -28,6 +28,133 @@ class Salesforce {
 		'webhook_id'    => '',
 	];
 
+	/**
+	 * Add hooks.
+	 */
+	public static function init() {
+		add_action( 'rest_api_init', [ __CLASS__, 'register_api_endpoints' ] );
+	}
+
+	/**
+	 * Register the endpoints needed to handle Salesforce sync webhooks.
+	 */
+	public static function register_api_endpoints() {
+		// Handle WooCommerce webhook to sync with Salesforce.
+		register_rest_route(
+			NEWSPACK_API_NAMESPACE,
+			'/salesforce/sync',
+			[
+				'methods'             => \WP_REST_Server::EDITABLE,
+				'callback'            => [ __CLASS__, 'api_sync_salesforce' ],
+				'permission_callback' => '__return_true',
```

Comment:
If this is only used as a WooCommerce webhook, is there a way to ensure this endpoint is not explored by other sources?

Reply:
395b52b validates the webhook ID sent as part of the request before executing the sync. It also handles both the new Salesforce API namespace (needed for settings-related endpoints in order to use the `PluginSettings` component) as well as the old API namespace.